### ISSUE TITLE
TextEditor.findMarkers returns all markers related to TextBuffer, not just those related to calling editor

### ISSUE THREAD

**Issue by lierdakil** [atom/atom#10994]
This is relevant in case there are multiple `TextEditor`s with the same `TextBuffer` open, e.g. after `split-...` command.

I'm not sure if this is a bug or a feature, but I'd prefer to get *only* markers related to calling `TextEditor`. Would that be possible without horrible hacks?

**Comment by abe33** [comment]
It depends, are you searching for markers you have created or for markers created by another package?

In the first case you probably want to use a marker layer on the text editor (through `TextEditor::addMarkerLayer`). In the second case, I don't see any mean to achieve that without modifying the package.

**Comment by lierdakil** [comment]
`addMarkerLayer` is marked as experimental though and as such is subject to change or even removal =\

Also, I found a (probably related) problem with markers disappearing. Not sure if I should file another issue.

Here's a gif:
![markers](https://cloud.githubusercontent.com/assets/7275622/13393060/219a44b2-def0-11e5-91bd-2a331215d10d.gif)

**Comment by nathansobo** [comment]
@lierdakil This change is actually coming with an overhaul of our approach to displaying text going on in this [branch of the text-buffer component](https://github.com/atom/text-buffer/tree/display-layers). I'm bracing myself for unexpected fallout from this nuanced change, but I agree that it's necessary for a variety of reasons.

For now, you should use marker layers if you can. The API *might* change slightly but at this point I'd say that's increasingly unlikely and we'll probably drop the experimental disclaimer once we land the above-mentioned changes. We definitely won't be removing the feature entirely.

**Comment by lee-dohm** [comment]
@lierdakil If that didn't answer your question or you have some functionality you would like to see, please let us know!

**Comment by lierdakil** [comment]
Okay... What about disappearing markers bug? Any insight? Should I file another issue?

**Comment by lee-dohm** [comment]
Please file another issue, preferably with repro steps :grinning:
